### ISSUE TITLE
Fix false positives for universal selector in no-descending-specificity

### ISSUE THREAD

**Issue by yukipastelcat** [stylelint/stylelint#3738]
```scss
.something {
  & > * {
    &:not(:first-child) {
      margin-right: 16px;
    }
    &:not(:last-child) {
      margin-left: 16px;
    }
  }
  &-element {
    & > * { /*Expected selector ".something-element > *" to come before selector ".something > *:not(:first-child)" (no-descending-specificity) */
      &:not(:first-child) {
        margin-top: 16px;
      }
      &:not(:last-child) {
        margin-bottom: 16px;
      }
    }
  }
}
```

My stylelint.config.js:
```javascript
module.exports = {
  extends: ['stylelint-config-recommended-scss'],
};
```

**Comment by jeddy3** [comment]
@yukipastelcat Thanks for the report.

> Why am I getting no-descending-specificity error?

Technically `.something-element > *` has a specificity of `0,1,0` and `.something > *:not(:first-child)` has a higher specificity of `0,2,0`, yet `.something-element > *` appears after `.something > *:not(:first-child)` in the source order. The rule is comparing them because I believe the last _compound selector_ in each selector match i.e. it's the _universal selector_. I can see how this could be undesirable because:

a. `& > * { }` has no declarations
b.  the universal selector is a odd thing to match

I believe this might be duplicate of https://github.com/stylelint/stylelint/issues/2489. It's proving a difficult problem to crack and this is proving to be a problematic rule, so please help in that issue if it's related or here if you think it's a separate one.

**Comment by Mouvedia** [comment]
@yukipastelcat I cannot reproduce your error on the latest version of Stylelint.
This was fixed by #7701; thanks @romainmenke.

[demo](https://stylelint.io/demo/#N4Igxg9gJgpiBcIB0BnCBbGAXAFgSwDsBzAAmAB0CSSAyEgPhICozLrqb4CIsAKeAGZ4ATiiwBaMPgA2UAJSsq7augCGwooXHC8RHFngkAjADYADgA8A3G3YBfWxy49+01WMkz5i5SvWaCcWkYAQNjc2tHEgclGI5xGGDMAiwfdjpGFgolZU5uPkERDyk8WQVs33Y1DS0sCDNDU0sbHPsop3zXdwkSsrTK6oDxACMeOvRGiJbKuLbYyjsQABoQIWCAOVVMBBAYCy2zYKQwFBRl8AgCISIdipJyXYssGAIoFAfDAG0HsQBPYOkhB6l2u2hgkHQyVgUHEKBO7xAAF0FudIFddAAxCDCNRYHYAKzQBHOsDMZ0Qdx+WH+iSBH3uIFMSBMSAADA8lrYqTTASlJCDdGCIVCYDC4ad6Q83M8xA8UXYgA)
